packages/yew/src/html/component/lifecycle.rs

```diff
@@ -302,10 +302,12 @@ impl ComponentState {
         if let Some(m) = self.suspension.take() {
             let comp_scope = self.inner.any_scope();
 
-            let suspense_scope = comp_scope.find_parent_scope::<BaseSuspense>().unwrap();
-            let suspense = suspense_scope.get_component().unwrap();
-
-            suspense.resume(m);
```

Comment:
We don't need to acquire a `Ref` to the Suspense component if all it does is to send a message to itself.

If the race condition happened on this line, it can be fixed by turning resume into an associated function (`Suspense::resume(suspense_scope, m)`).